### ISSUE TITLE
Git Desktop Authentication Issue (Mac OS)

### ISSUE THREAD

**Issue by manieldack** [desktop/desktop#18744]
### The problem

I know there are a couple of other issues with similar problems (and same error pop up) such as Authentication issue #18693. Due to the different OS, and the fact I am using Github as the repository hub, I figured this might warrant a new issue. 

I am using an intel Mac and in the last day, I could pull changes to repos where I am an admin level user, but could no longer push commits. I upgraded to v 3.4.0, and have signed out and signed back in a couple of different times and have gotten the exact same error. 

I recently tried a CL push and got an authentication issue there as well about write access. Verified that I am indeed an admin for this particular repo.  

### Release version

3.4.0

### Operating system

macOS 11.7.10

**Comment by sergiou87** [comment]
Could you upload the log file from GitHub Desktop so that I could get some more information about this error? To access the log files go to the file menu in GitHub Desktop and select Help > Show Logs.
The log files are created daily -- please upload a log file as an attachment from a day where you experienced the issue.

**Comment by marioscialpi** [comment]
same issue on apple silicon

logs:
```
2024-06-03T10:25:09.688Z - error: [ui] fetchMetadata: unable to load metadata from 'https://gitlab.xxxx.com/api/v3/meta' as a fallback
Error: API error https://gitlab.xxxx.com/api/v3/meta:  (410)
    at pn (/webpack:/app/src/lib/http.ts:178:11)
    at /webpack:/app/src/lib/api.ts:2066:11
    at Hp.endpointSupportsBasicAuth (/webpack:/app/src/lib/stores/sign-in-store.ts:255:11)
    at Hp.setEndpoint (/webpack:/app/src/lib/stores/sign-in-store.ts:549:13)
```
url @ 'https://gitlab.xxxx.com/api/v3/meta' return this json:

`{"error":"API V3 is no longer supported. Use API V4 instead."}`

**Comment by manieldack** [comment]
I just tried a push and got the same error, so this is the log from today

[2024-06-03.desktop.production.log](https://github.com/user-attachments/files/15537295/2024-06-03.desktop.production.log)


2024-06-03T00:11:59.692Z - info: [ui] [BranchPruner] Last prune took place 8 hours ago - skipping
2024-06-03T04:12:00.116Z - info: [ui] [BranchPruner] Last prune took place 12 hours ago - skipping
2024-06-03T06:26:00.598Z - info: [ui] Stats reported.
2024-06-03T08:12:00.586Z - info: [ui] [BranchPruner] Last prune took place 16 hours ago - skipping
2024-06-03T12:12:01.230Z - info: [ui] [BranchPruner] Last prune took place 20 hours ago - skipping
2024-06-03T16:12:01.770Z - info: [ui] [BranchPruner] No branches to prune.
2024-06-03T17:05:00.956Z - error: [ui] `git -c credential.helper= push origin master:master --progress` exited with an unexpected code: 128.
stderr:
ERROR: Write access to repository not granted.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

(The error was parsed as 3: Authentication failed. Some common reasons include:

- You are not logged in to your account: see GitHub Desktop > Settings.
- You may need to log out and log back in to refresh your token.
- You do not have permission to access this repository.
- The repository is archived on GitHub. Check the repository settings to confirm you are still permitted to push commits.
- If you use SSH authentication, check that your key is added to the ssh-agent and associated with your account.
- If you use SSH authentication, ensure the host key verification passes for your repository hosting service.
- If you used username / password authentication, you might need to use a Personal Access Token instead of your account password. Check the documentation of your repository hosting service.)
2024-06-03T17:05:03.817Z - info: [ui] Executing fetch: git -c credential.helper= fetch --progress --prune --recurse-submodules=on-demand origin (took 1.040s)
2024-06-03T17:05:04.563Z - info: [ui] [RepositoryIndicatorUpdater]: Refreshing sidebar indicators for 25 repositories took 6.7s of which 0.0s paused, total 6.7s

**Comment by sergiou87** [comment]
@manieldack did you clone it via SSH or HTTPS? Could you share the clone URL? (You can see it in the `Repository -> Repository Settings…` menu) Feel free to redact any parts of it that you don't wanna share.

**Comment by sergiou87** [comment]
Another question: does the app prompt you for credentials of that repo at any point?

**Comment by manieldack** [comment]
This is the clone URL from that command: 

git@github.com:kcrbba/reponame.git

It never prompts me for credentials. Worth noting that I am an admin of that organization which is where we have organized my company's repos.

**Comment by manieldack** [comment]
[2024-06-07.desktop.production.log](https://github.com/user-attachments/files/15743851/2024-06-07.desktop.production.log)

I have uploaded another log, this time after being prompted for the 3.4.1 update, and then did a full signout and sign back in, and still getting the same error, in case that brings more insight.

**Comment by steveward** [comment]
@manieldack thanks for the log. It looks this is a permissions issue for this repository, since you encountered a write access error from the CLI as well as in GitHub Desktop. Are you an owner of the org, or just an admin of a team with access to the repository? I would recommend verifying the permissions are correct again, since there is quite a bit of granularity. You can check this by going to `https://github.com/<org_name>/<repo_name>/settings/access`.

**Comment by manieldack** [comment]
Makes sense. Just went to verify. I am an owner of the organization, and for that particular repo, I had write access. I changed it to admin, and still encountered the same error on the desktop.

**Comment by marioscialpi** [comment]
> Another question: does the app prompt you for credentials of that repo at any point?

in my case, yes, the app prompt for credentials.

I downloaded an older version (3.3.18) and this issue is not present, nor app prompt for credentials

New version 3.4.1 does not solve

**Comment by niik** [comment]
Hey @manieldack, @marioscialpi, we just shipped GitHub Desktop 3.4.2-beta2 to our [beta channel](https://desktop.github.com/beta/). This includes support for using [Git Credential Manager](https://gh.io/gcm). It's currently off by default but you can enable it in Settings under the advanced tab. We believe this will address the authentication problems outlined in this issue.

<img width="608" alt="Screenshot of Desktop's advanced preferences section with a highlight around the new option to enable Git Credential Manager" src="https://github.com/desktop/desktop/assets/634063/5d4da983-7663-49dd-bb7b-ce51ee9d2d3a" />

When Git Credential Manager (GCM) is enabled all credentials for third-party (i.e. Azure Devops, BitBucket, GitLab etc) will be handled, and stored, outside of GitHub Desktop. GCM supports browser-sign in to many popular services and will avoid the need to create personal access tokens (PATs). 

Please download our beta version from https://desktop.github.com/beta/ to give this a go and please report back with your experience. We'll be collecting feedback in our beta channel which will inform when we make this available to all users. Installing the beta version will not clear your preferences or tracked repositories and you can switch to the production version by simply downloading the latest version from https://desktop.github.com.

**Comment by steveward** [comment]
@manieldack if you're still encountering this issue, could you try cloning a private repository from your personal account? That would help narrow down if this is a specific org-level permission issue and not something else.
